src/index.js

```diff
@@ -4,59 +4,128 @@ import { sortArrayByName } from './utils';
-            const renderedDetails = renderMovieDetails(moviesData[movie_id]);
-            movieDetailsContainer.append(renderedDetails);
-            clearMovies(movieWrap);
-        }
+        unregisterFromClickOn();
+        unregisterFromMouseOver();
+    }
+}
 
-    }, false);
+const networkObserver = new EventObserver();
+const moviesArr = [];
 
-    const latest_mouseover_obj = {
-        class: '',
-        id: ''
-    };
+const latest_mouseover_obj = {
+    class: '',
+    id: ''
+};
 
-    movieContainerDiv.addEventListener('mouseover', (evt) => {
-        let className = evt.target.getAttribute('class');
+// mouseover
+function handleMouseOver(evt) {
+    const className = evt.target.getAttribute('class');
 
         if (className === 'movie__card') {
             const modal = document.getElementById("myModal");
             const movie_id = evt.target.getAttribute('id');
 
-            if (   latest_mouseover_obj.class !== "movie__card"
+            if (latest_mouseover_obj.class !== "movie__card"
                 || (latest_mouseover_obj.class === "movie__card" && latest_mouseover_obj.id !== movie_id)) {
 
-                var p = document.getElementById("modal__text");
-                p.innerHTML = `${moviesData[movie_id].description}`;
+                const movie_card = document.getElementById(movie_id);
+                const movie_card_children = movie_card.childNodes;
 
-                modal.style.display = "block";
+                const movie_popup = [].find.call(movie_card_children, elem => (elem.className === "movie__popup"))
 
-                setTimeout(function () {
-                    modal.style.display = "none";
-                }, 1500);
+                if (movie_popup !== undefined) {
+                    movie_popup.style.display = "block";
+                    setTimeout(function () {
+                        movie_popup.style.display = "none";
+                    }, 3000);
+                }
 
                 latest_mouseover_obj.id = movie_id;
             } else {
                 latest_mouseover_obj.id = '';
             }
         }
         latest_mouseover_obj.class = className;
+}
 
-    }, false);
+
+function registerForClickOn() {
+    const movieContainerDiv = getMovieContainerDiv();
+    movieContainerDiv.addEventListener('click', handleClickOnMovie);
+}
+
+function unregisterFromClickOn() {
+    const movieContainerDiv = getMovieContainerDiv();
+    movieContainerDiv.removeEventListener('click', handleClickOnMovie);
 }
 
+function registerForMouseOver() {
+    const movieContainerDiv = getMovieContainerDiv();
+    movieContainerDiv.addEventListener('mouseover', handleMouseOver);
+}
+
+function unregisterFromMouseOver() {
+    const movieContainerDiv = getMovieContainerDiv();
+    movieContainerDiv.removeEventListener('mouseover', handleMouseOver);
+}
+
+function handleError() {
+    //getMovieContainerDiv().innerText = '⚠️';
```

Comment:
Do not commit unused code (stash it locally or make separate branch, if you have a lot of temporary changes)